src/compiler/checker.ts

```diff
@@ -16420,6 +16426,14 @@ namespace ts {
             return suggestion && symbolName(suggestion);
         }
 
+        function typeHavingProperty(type: Type, property: __String, errorNode?: Node): Type | undefined {
+            const types: Type[] = (type.flags & TypeFlags.UnionOrIntersection)

```

Comment:
i do not think you want to look in unions or intersections. if the promised type has the property then show the message.. not if one of the constituents of the union type has it.. since `await` will not fix this problem.